riscv-executor/src/lib.rs

```diff
@@ -807,10 +854,10 @@ impl<'a, 'b, F: FieldElement> Executor<'a, 'b, F> {
 pub fn execute_ast<T: FieldElement>(
     program: &AnalysisASMFile<T>,
     inputs: &Callback<T>,
-    bootloader_inputs: &[T],
+    bootloader_inputs: &[Elem<T>],
```

Comment:
Since `Elem` is only used here, wouldn't it be easier to take in `&[T]` and build the `Elem`s here?

Reply:
I did like this to preserve the type across chunk executions. I.e. imagine an execution was split between a `poseidon_gl` and a `split_gl`. The "fieldness" of the register containing the poseidon result must be preserved.